Code/Tools/ProjectManager/Source/CreateAGemScreen.cpp

```diff
@@ -0,0 +1,511 @@
+        if (m_secondDropdown->comboBox()->currentIndex() != 0 && m_secondDropdown->comboBox()->currentIndex() ==
+            m_firstDropdown->comboBox()->currentIndex())
+        {
+            firstOptionalGemTagIsValid = false;
+            m_secondDropdown->setErrorLabelText(tr("This gem has already been selected"));
+        }
+
+        if (m_thirdDropdown->comboBox()->currentIndex() != 0 && (m_thirdDropdown->comboBox()->currentIndex() == m_secondDropdown->comboBox()->currentIndex() ||
+            m_thirdDropdown->comboBox()->currentIndex() == m_firstDropdown->comboBox()->currentIndex()))
+        {
+            secondOptionalGemTagIsValid = false;
+            m_thirdDropdown->setErrorLabelText(tr("This gem has already been selected"));
+        }
+
+        m_secondDropdown->setErrorLabelVisible(!firstOptionalGemTagIsValid);
+        m_thirdDropdown->setErrorLabelVisible(!secondOptionalGemTagIsValid);
+        return firstOptionalGemTagIsValid && secondOptionalGemTagIsValid;
+    }
+
+
+    bool CreateAGemScreen::ValidateCreatorName()
+    {
+        bool creatorNameIsValid = true;
+        if (m_creatorName->lineEdit()->text().isEmpty())
+        {
+            creatorNameIsValid = false;
+            m_creatorName->setErrorLabelText(tr("A creator's name is required."));
+        }
+        m_creatorName->setErrorLabelVisible(!creatorNameIsValid);
+        return creatorNameIsValid;
+    }
+
+    bool CreateAGemScreen::ValidateRepositoryURL()
+    {
+        bool repositoryURLIsValid = true;
+        if (m_repositoryURL->lineEdit()->text().isEmpty())
+        {
+            repositoryURLIsValid = false;
+            m_repositoryURL->setErrorLabelText(tr("A repository URL is required."));
+        }
+        m_repositoryURL->setErrorLabelVisible(!repositoryURLIsValid);
+        return repositoryURLIsValid;
+    }
+
+    void CreateAGemScreen::HandleBackButton()
+    {
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            m_tabWidget->setCurrentIndex(1);
+            m_nextButton->setText(tr("Next"));
+        }
+        else if (m_tabWidget->currentIndex() == 1)
+        {
+            m_backButton->setVisible(false);
+            m_tabWidget->setCurrentIndex(0);
+        }
+        else if (m_tabWidget->currentIndex() == 0)
+        {
+            m_backButton->setVisible(false);
+        }
+    }
+
+    void CreateAGemScreen::HandleNextButton()
+    {
+        if (m_tabWidget->currentIndex() == 0)
+        {
+            bool templateLocationIsValid = ValidateGemTemplateLocation();
+            if (!m_formFolderRadioButton->isChecked())
+            {
+                m_backButton->setVisible(true);
+                m_tabWidget->setCurrentIndex(1);
+            }
+            else if (templateLocationIsValid && m_formFolderRadioButton->isChecked())
+            {
+                m_backButton->setVisible(true);
+                m_tabWidget->setCurrentIndex(1);
+            }
```

Comment:
something about this logic seems odd
```suggestion
            if (!m_formFolderRadioButton->isChecked() || ValidateGemTemplateLocation())
            {
                m_backButton->setVisible(true);
                m_tabWidget->setCurrentIndex(1);
            }
```

Reply:
ValidateGemTemplateLocation returns true if the specific folder selection isn't checked anyway, I've simplified this further.